FEATURE_PROCESS.adoc

```diff
@@ -1,5 +1,106 @@
 = WildFly Feature Development Process
 
+The WildFly Feature Development Process defines the procedure that must be followed during the _development_ of features that appear in any of the following:
+
+* The standard WildFly distribution.
+* The WildFly quickstarts.
+* The WildFly Docker and OpenShift images.
+* Standard end-user tooling documented for use with standard WildFly.
+* WildFly Preview, if the feature is at `community` stability or higher.
+* Feature packs that WildFly Glow suggests to users. (WildFly Glow may introduce the concept of "incubating" feature packs that users can opt into having Glow suggest. Development of such feature packs need not follow this process, but they *must not* provide features at `community` or higher stability).
+
+This process should be followed once a developer anticipates beginning work on a feature in the next three months.
+
+NOTE: Requesting new features or recording ideas for new features are *not* part of this process. Feature requests are handled by simply filing an issue in the relevant project JIRA, typically https://issues.redhat.com/projects/WFLY/summary[WFLY].
+
+== The Process
+
+Feature development proceeds in five stages:
+
+=== Kickoff
+
+When a developer intends to begin work on a feature they should do the following:
+
+* If one doesn't already exist, file an issue in the relevant project JIRA, typically https://issues.redhat.com/projects/WFLY/summary[WFLY], https://issues.redhat.com/projects/WFCORE/summary[WFCORE] or https://issues.redhat.com/projects/HAL/summary[HAL].
+* File an issue with the https://github.com/orgs/wildfly/projects/7[WildFly Feature Planning project]. To do this create an issue using the https://github.com/wildfly/wildfly-proposals/issues/new?assignees=&labels=feature&projects=wildfly%2F7&template=feature-development.yaml[Feature Development issue template].
+* Draft a minimal feature analysis, based on the current https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc[design-doc-template]. This should include the expected stability level, links to the issues above, and enough Overview and User Story information to give people interested in being part of the feature team a sense of what would be involved.
+* Submit a draft analysis PR to the https://github.com/wildfly/wildfly-proposals[`wildfly-proposals`] repository.
+* Send an email to the `wildfly-dev` mail list telling people about the upcoming work and asking for volunteers to participate in the feature team.
```

Comment:
IMO the wildfly-dev list should be the full address, I am not sure if non subscribers can post so if not maybe we should also point to sign up information here.

Reply:
+1 to the full mail address. I was too lazy to look it up when I drafted this. I'll add the sign up link.

Reply:
Done.

Reply:
A very minor point under your discretion. Could we also add an email subject format here? It could be useful to anyone who wants to filter and classify these message using the email client.

Following what has been sent in the past, the subject could be:

"Feature Team Formation - [stability level] [Jira issue] Short description"

Reply:
If we want something to be in a specific form maybe it could be a part of automation on issue creation to send the initial message?

Reply:
If we do anything like this, let's use automation. I'd like to keep this as free of small details as possible so people might actually read it. :)

Also, these emails are good candidates for starting threads for continued discussion of the feature itself and not just the process.